src/authorization/AclOwnershipReader.ts

```diff
@@ -0,0 +1,52 @@
+import { CredentialGroup } from '../authentication/Credentials';
+import type { AccountSettings, AccountStore } from '../identity/interaction/email-password/storage/AccountStore';
+import type { AuxiliaryIdentifierStrategy } from '../ldp/auxiliary/AuxiliaryIdentifierStrategy';
+import type { AclPermission } from '../ldp/permissions/AclPermission';
+import type { PermissionSet } from '../ldp/permissions/Permissions';
+import type { PermissionReaderInput } from './PermissionReader';
+import { PermissionReader } from './PermissionReader';
+
+/**
+ * Allows full access on ACL resources if the request is being made by the owner of the pod containing that resource.
+ */
+export class AclOwnershipReader extends PermissionReader {
+  private readonly accountStore: AccountStore;
+  private readonly aclStrategy: AuxiliaryIdentifierStrategy;
+
+  public constructor(accountStore: AccountStore, aclStrategy: AuxiliaryIdentifierStrategy) {
+    super();
+    this.accountStore = accountStore;
+    this.aclStrategy = aclStrategy;
+  }
+
+  public async handle({ credentials, identifier }: PermissionReaderInput): Promise<PermissionSet> {
+    if (!this.aclStrategy.isAuxiliaryIdentifier(identifier)) {
+      // Exception is only granted when accessing ACL resources
+      return {};
+    }
+    if (!credentials.agent?.webId) {
+      // Only authenticated agents are supported
+      return {};
+    }
+    let settings: AccountSettings;
+    try {
+      settings = await this.accountStore.getSettings(credentials.agent.webId);
```

Comment:
If too expensive (re: performance above), maybe we can consider a cache.

Reply:
It might be interesting to have some sort of cache for the key/value storages. Did we already have a discussion about this? Because I feel like we did. We would need events to update the caches then. Might even allow us to use an N3 store for the data and have events contain info which data to add/remove.

Reply:
Yeah we did indeed. But not a necessity then; we know we can.